build/mvn

```diff
@@ -21,7 +21,12 @@ SELF=$(cd $(dirname $0) && pwd)
 . "$SELF/util.sh"
 
 if [ -z "${JAVA_HOME}" -a "$(command -v javac)" ]; then
-  export JAVA_HOME="$(dirname $(dirname $(realpath $(command -v javac))))"
+  if [ x"$(uname -a | awk '{print $1}')" = xDarwin -a \
+      $(version $(sw_vers -productVersion)) -ge $(version 10.5) ]; then
```

Comment:
Why not just check the existence of `/usr/libexec/java_home` rather than the version of macOS?

Reply:
I think just checking the existence of `/usr/libexec/java_home` is not a good idea, this path is not a standard for every OS, some one may put another command with same path but providing different behavior. So it's better to use a command provided by some specific OS obviously.

Reply:
Hmm, as you say, `/usr/libexec/java_home` is not standard path and user might put a file on the same path.
But, even though macOS 10.5+ is used, user still might overwrite the path with their own file.

Even if we check that the OS is macOS, it's too much to check the version. AFAIK Hadoop and Netty uses `/usr/libexec/java_home` for macOS but they just check whether `Darwin` or not.
Also, you can use `name -s` to get the OS name.

Reply:
According to the [link](https://mkyong.com/java/how-to-set-java_home-environment-variable-on-mac-os-x/#what-is-usrlibexecjava-home), `/usr/libexec/java_home` is a standard definition for macOS 10.5+.
We should assume that user should not destroy their OS environment by replacing a standard command by another file.

As you mentioned, `uname -s` is better than my solution, I have change my codes.

Reply:
Java 8 supports OS X 10.8.3+ and 10.9+.
https://java.com/download/help/sysreq.html

So, I don't think it's necessary to check whether the version is 10.5+ or not as it's trivial.

Reply:
Well, it's a different problem with what you said. I mean, `/usr/libexec/java_home` is an OS behavior which provides a command to output the home path of Java, it's irrelevant with the problem that what version of macOS is Java8 supported.

In other words, there are almost no one using mac OS with version less than 10.5 according to the [statistic](https://www.statista.com/statistics/944559/worldwide-macos-version-market-share/), so it's not important already to check whether version is 10.5+ or not. So, I can remove it if you insist.

Reply:
Yes, you are right.

Reply:
> it's irrelevant with the problem that what version of macOS is Java8 supported.

Building Spark with Maven requires Java 8 (https://spark.apache.org/docs/latest/building-spark.html#apache-maven).
Java 8 supports OS X 10.8.3+ and 10.9+.
So, someone who want to build with `build/mvn` should use 10.8.3+ or 10.9+.

Reply:
Yes, I realized this point just after I submitted the comment, and I deleted that comment immediately.